src/main/java/nl/hsac/fitnesse/fixture/fit/SoapCallMapColumnFixture.java

```diff
@@ -131,4 +141,32 @@ protected String xmlAsHtml(String xml) {
         return getEnvironment().getHtmlForXml(xml);
     }
 
+    private void correctResponse(XmlHttpResponse response) {
+        try {
+            // Avoid exceptions when there are no multiparts
+            System.setProperty("mail.mime.multipart.allowempty", "true");
+            MimeMultipart mp = new MimeMultipart(new ByteArrayDataSource(response.getResponse(), "application/soap+xml"));
+
+            int count = mp.getCount();
+            if (count > 0) {
+                String xml = getBodypartString(mp.getBodyPart(0));
+                response.setResponse(xml);
+            }
+        } catch (MessagingException | IOException e) {
+            throw new RuntimeException("unparseble response", e);
+        }
+    }
+
+    private String getBodypartString(BodyPart bp) throws IOException, MessagingException {
+        SharedByteArrayInputStream stream = (SharedByteArrayInputStream) bp.getContent();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(stream))) {
```

Comment:
Isn't there a convenient method in the apache libraries (e.g. IOUtils) to copy a stream to a string?